src/Sulu/Bundle/MediaBundle/Resources/js/containers/MediaSelection/MediaSelection.js

```diff
@@ -0,0 +1,283 @@
+    }
+
+    @action setCollectionId(id: ?string | number) {
+        this.collectionId = id;
+    }
+
+    destroyOverlay() {
+        this.selectedMediaIds = [];
+        this.collectionId = undefined;
+
+        this.destroyStores();
+        this.overlayDisposer();
+    }
+
+    createStores = () => {
+        this.setMediaPage(1);
+        this.setCollectionPage(1);
+
+        this.createCollectionStore(this.collectionId, this.props.locale);
+        this.createMediaDatagridStore(this.collectionId, this.mediaPage, this.props.locale);
+        this.createCollectionDatagridStore(this.collectionId, this.collectionPage, this.props.locale);
+    };
+
+    destroyStores() {
+        if (this.collectionStore) {
+            this.collectionStore.destroy();
+        }
+
+        if (this.mediaDatagridStore) {
+            this.mediaDatagridStore.destroy();
+        }
+
+        if (this.collectionDatagridStore) {
+            this.collectionDatagridStore.destroy();
+        }
+    }
+
+    @action createCollectionDatagridStore(collectionId: ?observable, page: observable, locale: string) {
+        if (this.collectionDatagridStore) {
+            this.collectionDatagridStore.destroy();
+        }
+
+        this.collectionDatagridStore = new DatagridStore(
+            COLLECTIONS_RESOURCE_KEY,
+            {
+                page,
+                locale,
+            },
+            (collectionId) ? {parent: collectionId} : undefined
+        );
+    }
+
+    createCollectionStore = (collectionId: ?observable, locale: string) => {
+        if (this.collectionStore) {
+            this.collectionStore.destroy();
+        }
+
+        this.collectionStore = new CollectionStore(collectionId, locale);
+    };
+
+    @action createMediaDatagridStore(collectionId: ?observable, page: observable, locale: string) {
+        const options = {};
+
+        options.fields = [
+            'id',
+            'type',
+            'name',
+            'size',
+            'title',
+            'mimeType',
+            'subVersion',
+            'thumbnails',
+        ].join(',');
+
+        page.set(1);
+
+        if (collectionId) {
+            options.collection = collectionId;
+        }
+
+        if (this.mediaSelectionStore.selectedMediaIds.length) {
+            options.exclude = this.mediaSelectionStore.selectedMediaIds.join(',');
+        }
+
+        if (this.mediaDatagridStore) {
+            this.mediaDatagridStore.destroy();
+        }
+
+        this.mediaDatagridStore = new DatagridStore(
+            MEDIA_RESOURCE_KEY,
+            {
+                page,
+                locale,
+            },
+            options,
+            true,
+            this.handleMediaSelection,
+            this.selectedMediaIds
+        );
+    }
+
+    addMediaToSelectionStore() {
+        this.selectedMediaIds.forEach((mediaId) => {
+            const media = this.mediaDatagridStore.data.find((item) => item.id === mediaId);
```

Comment:
I think that can't really work... What if you've added a media, change the collection afterwards, and press ok then? Then the store you are accessing here wouldn't have the item you are looking for anymore.